core/src/main/scala/kafka/log/Log.scala

```diff
@@ -208,7 +208,12 @@ class Log(@volatile var dir: File,
       .format(name, segments.size(), logStartOffset, logEndOffset, time.milliseconds - startMs))
   }
 
-  private val tags = Map("topic" -> topicPartition.topic, "partition" -> topicPartition.partition.toString)
+  private val tags = {
+    if (isFuture)
+      Map("topic" -> topicPartition.topic, "partition" -> topicPartition.partition.toString, "isFuture" -> "true")
+    else
+      Map("topic" -> topicPartition.topic, "partition" -> topicPartition.partition.toString)
```

Comment:
It seems like `if/else` could be about appending the third tag instead of duplicating all the logic.

Reply:
Sure. I updated the patch with the following code:

```
private val tags = {
  val maybeFutureTag = if (isFuture) Map("is-future" -> "true") else Map.empty[String, String]
  Map("topic" -> topicPartition.topic, "partition" -> topicPartition.partition.toString) ++ maybeFutureTag
}
```